cassandra-mesos-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/executor/jmx/NodeBackupJob.java

```diff
@@ -72,11 +77,13 @@ public void run() {
                     LOGGER.info("Starting backup on keyspace {}", keyspace);
                     keyspaceStarted();
 
-                    checkNotNull(jmxConnect).getStorageServiceProxy().takeSnapshot(backupName, keyspace);
-                    keyspaceFinished("SUCCESS", keyspace);
+                    final BackupManager backupManager = new BackupManager(checkNotNull(jmxConnect), backupDir);
+                    backupManager.backup(keyspace);
+
+                    keyspaceFinished(SUCCESS, keyspace);
                 } catch (final Exception e) {
                     LOGGER.error("Failed to backup keyspace " + keyspace, e);
-                    keyspaceFinished("FAILURE", keyspace);
```

Comment:
Okay, I think this approach is reasonable as long as we fail fast (stop processing the keyspace and move onto the next one). Also, is the user notified of this failure? Are then able to try again with the specific keyspace? Just wanted to clarify